### PR TITLE
Pass access scopes on query string

### PR DESCRIPTION
**Pull request by andyw8**
### What this PR does

See explanation in related PR: https://github.com/Shopify/shopify-api-ruby/pull/1023

### Reviewer's guide to testing

Use dev version of https://shopify-graphiql-app.shopifycloud.com/

### Checklist

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [x] ~Update `README.md`, if appropriate.~
- [x] ~Update any relevant pages in `/docs`, if necessary~
- [x] ~For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.~

### CODE REVIEW

File: lib/shopify_app/controller_concerns/login_protection.rb [hunk 1]
```diff
@@ -167,6 +167,10 @@ def login_url_params(top_level:)
         query_params[:host] ||= host
       end
 
+      if params[:access_scopes].present?
+        query_params[:scope] = params[:access_scopes].join(",")
```
Comment (andyw8) [inline lib/shopify_app/controller_concerns/login_protection.rb:171]: Other places in this method use `||=` but I don't see a need for that here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
+* Pass access scopes on query string [#1540](https://github.com/Shopify/shopify_app/pull/1540)
+
 21.1.1 (Oct 20, 2022)
 ----------
 * Updates dependency to `shopify_api` to 12.2 to fix error with host_name argument.
 
 21.1.0 (Oct 17, 2022)
 ----------
-
```
Comment (andyw8) [inline CHANGELOG.md:10]: Removed these for consistency.

File: lib/shopify_app/controller_concerns/login_protection.rb (hunk 1 above)
Comment (paulomarg) [reply lib/shopify_app/controller_concerns/login_protection.rb:171]: Makes sense to me, if we get query args we should override them, assuming we can trust the source - is this secured in any way, like with an HMAC?

**Review by paulomarg — APPROVED** [review]
Just a general question, but if there is nothing to change then I'm happy with this.

**Review by nelsonwittwer — APPROVED** [review]

File: lib/shopify_app/controller_concerns/login_protection.rb (hunk 1 above)
Comment (andyw8) [reply lib/shopify_app/controller_concerns/login_protection.rb:171]: (Discussed internally, we do not see any issue here)

**Pull request merged**